internal/lookoutingesterv2/instructions/instructions.go

```diff
@@ -0,0 +1,553 @@
+		return errors.WithStack(err)
+	}
+
+	for _, e := range event.GetErrors() {
+		// We just interpret the first terminal error
+		if e.Terminal {
+
+			// Certain legacy events mean we don't have a valid run id
+			// In this case we have to invent a fake run
+			// TODO: remove this when the legacy messages go away!
+			isLegacyEvent := runId == eventutil.LEGACY_RUN_ID
+			if isLegacyEvent {
+				jobRun := createFakeJobRun(jobId, ts)
+				runId = jobRun.RunId
+				objectMeta := extractMetaFromError(e)
+				if objectMeta != nil && objectMeta.ExecutorId != "" {
+					jobRun.Cluster = objectMeta.ExecutorId
+				}
+				update.JobRunsToCreate = append(update.JobRunsToCreate, jobRun)
+			}
+
+			jobRunUpdate := &model.UpdateJobRunInstruction{
+				RunId:    runId,
+				Finished: &ts,
+			}
+			if isLegacyEvent {
+				jobRunUpdate.Started = &ts
+			}
+
+			switch reason := e.Reason.(type) {
+			case *armadaevents.Error_PodError:
+				jobRunUpdate.Node = extractNodeName(reason.PodError)
+				jobRunUpdate.JobRunState = pointer.Int32(database.JobRunFailedOrdinal)
+				jobRunUpdate.Error = tryCompressError(jobId, reason.PodError.GetMessage(), c.compressor)
+				var exitCode int32 = 0
+				for _, containerError := range reason.PodError.ContainerErrors {
+					if containerError.ExitCode != 0 {
+						exitCode = containerError.ExitCode
+						break
+					}
+				}
+				jobRunUpdate.ExitCode = pointer.Int32(exitCode)
+			case *armadaevents.Error_PodTerminated:
+				jobRunUpdate.Node = extractNodeName(reason.PodTerminated)
+				jobRunUpdate.JobRunState = pointer.Int32(database.JobRunTerminatedOrdinal)
+				jobRunUpdate.Error = tryCompressError(jobId, reason.PodTerminated.GetMessage(), c.compressor)
+			case *armadaevents.Error_PodUnschedulable:
+				jobRunUpdate.Node = extractNodeName(reason.PodUnschedulable)
+				jobRunUpdate.JobRunState = pointer.Int32(database.JobRunUnableToScheduleOrdinal)
+				jobRunUpdate.Error = tryCompressError(jobId, reason.PodUnschedulable.GetMessage(), c.compressor)
+			case *armadaevents.Error_PodLeaseReturned:
+				jobRunUpdate.JobRunState = pointer.Int32(database.JobRunLeaseReturnedOrdinal)
+				jobRunUpdate.Error = tryCompressError(jobId, reason.PodLeaseReturned.GetMessage(), c.compressor)
+			case *armadaevents.Error_LeaseExpired:
+				jobRunUpdate.JobRunState = pointer.Int32(database.JobRunLeaseExpiredOrdinal)
+				jobRunUpdate.Error = tryCompressError(jobId, "Lease expired", c.compressor)
+			default:
+				jobRunUpdate.JobRunState = pointer.Int32(database.JobRunFailedOrdinal)
+				jobRunUpdate.Error = tryCompressError(jobId, "Unknown error", c.compressor)
+				log.Debugf("Ignoring event %T", reason)
+			}
+			update.JobRunsToUpdate = append(update.JobRunsToUpdate, jobRunUpdate)
+			break
+		}
+	}
+	return nil
+}
+
+func tryCompressError(jobId string, errorString string, compressor compress.Compressor) []byte {
+	compressedError, err := compressor.Compress([]byte(util.RemoveNullsFromString(errorString)))
```

Comment:
out of interest, why do we need to remove the nulls here?

Reply:
In my understanding, for the same reason we should remove it from the Job Proto (double checking now, ti seems that isn't happening despite the test being there). This caused problems at insertion time in Postgres when trying to store null chars, but maybe it is not relevant anymore for byte array fields. Will test to confirm

Reply:
Not needed, just works for byte arrays. Have removed